client/landing/stepper/declarative-flow/internals/steps.tsx

```diff
@@ -319,4 +319,10 @@ export const STEPS = {
 		slug: 'migration-upgrade-plan',
 		asyncComponent: () => import( './steps-repository/migration-upgrade-plan' ),
 	},
+
+	USER: {
+		slug: 'user',
+		asyncComponent: () =>
+			import( /* webpackChunkName: "stepper-user-step" */ './steps-repository/__user' ),
+	},
```

Comment:
We don't want the user step in the steps library. It's a private step that should only used by the framework itself.

Reply:
I don't think it makes sense because there are some components know what's the user step and use hardcoded slug, `user`, to determine whether it's a user step. It doesn't look like a private step. In addition, what's the benefit of making the user step private? If there is a flow that requires auth at the beginning, we can still make the user step to the first step explicitly.

Reply:
> what's the user step and use hardcoded slug.

Yes, its slug should be hardcoded and not up to the individual flows. If flows take control of adding the user step, they would be able to alter its slug. This would bring issues because the user step assumes its slug is `user` in OAuth redirects and such.

Also, Stepper will have the ability to optimize, like not loading the user step when the user is logged in. 

Stepper leaves most of the things to the flows, and that's a good thing, but now there is too much of freedom. Flows are making up their own `goToCheckout` and `exitFlow` utils etc.. With tons of copy-pasted code. I'm trying to gradually bring a nice and more opinionated API to unify things. The user step is the first baby steps towards that.

Reply:
> Yes, its slug should be hardcoded and not up to the individual flows. If flows take control of adding the user step, they would be able to alter its slug. This would bring issues because the user step assumes its slug is user in OAuth redirects and such.

I assume if flows want to add the user step manually, they will simply import the user step from this file. If they want to alter the slug, they will still make change to export the user step from the original file or hardcoded-adding it to their steps manually. IMHO, it would be great to avoid hardcoded slug here and there even if we want to keep it private. Or at least, a function to determine whether it's a user step. WDYT?

BTW there are still lots flows that define the steps by themselves instead of importing from pre-defined steps 😥

> Stepper leaves most of the things to the flows, and that's a good thing, but now there is too much freedom. Flows are making up their own goToCheckout and exitFlow utils etc.. With tons of copy-pasted code. I'm trying to gradually bring a nice and more opinionated API to unify things. The user step is the first baby step towards that.

Yes, it would be great if we could remove lots of copy-pasted code.

Reply:
> If they want to alter the slug, they will still make change to export the user step from the original file or hardcoded-adding it to their steps manually

Yes, but the step name starts with `__user` and its readme says please don't import. I really hope no one imports it manually.